suite-common/wallet-core/src/fiat-rates/fiatRatesThunks.ts

```diff
@@ -31,37 +41,89 @@ export const updateTxsFiatRatesThunk = createThunk(
     ) => {
         if (txs?.length === 0 || isTestnet(account.symbol)) return;
 
-        const timestamps = txs.map(tx => tx.blockTime!);
-
         const isElectrumBackend = selectIsElectrumBackendSelected(getState(), account.symbol);
 
         try {
+            const timestamps = txs.map(tx => tx.blockTime);
+            const roundedTimestamps = roundTimestampsToNearestPastHour(timestamps as Timestamp[]);
+            const uniqueTimestamps = [...new Set(roundedTimestamps)];
+
             const results = await getFiatRatesForTimestamps(
                 { symbol: account.symbol },
-                timestamps,
+                uniqueTimestamps,
                 localCurrency,
                 isElectrumBackend,
             );
 
             if (results && 'tickers' in results) {
                 dispatch(
-                    transactionsActions.updateTransactionFiatRate(
-                        txs.map((tx, i) => ({
-                            txid: tx.txid,
-                            updateObject: { rates: results.tickers[i]?.rates },
-                            account,
-                            ts: Date.now(),
+                    fiatRatesActions.addFiatRatesForTimestamps({
+                        ticker: { symbol: account.symbol },
+                        localCurrency,
+                        rates: results.tickers.map((ticker, index) => ({
+                            rate: ticker?.rates[localCurrency],
+                            lastTickerTimestamp: uniqueTimestamps[index],
                         })),
-                    ),
+                    }),
                 );
             }
         } catch (error) {
             console.error(error);
         }
+
+        const groupedTokensTxs = groupTokensTransactionsByContractAddress(txs);
+
+        for (const token in groupedTokensTxs) {
+            const hasCoinDefinitions = getNetworkFeatures(account.symbol as NetworkSymbol).includes(
+                'coin-definitions',
+            );
+            if (hasCoinDefinitions) {
+                const isTokenKnown = selectIsSpecificCoinDefinitionKnown(
+                    getState(),
+                    account.symbol as NetworkSymbol,
+                    token as TokenAddress,
+                );
+
+                if (!isTokenKnown) {
+                    throw new Error('Missing token definition');
```

Comment:
In case one token in the whole account is unknown, you do not fetch anything.
```suggestion
                    continue;
```